src/app/ServiceRegistry/components/ServiceRegistryMapping/ServiceRegistryMapping.tsx

```diff
@@ -60,22 +60,39 @@ export const ServiceRegistryMapping: React.FC<ServiceRegistryMappingProps> = ({
   }, [selectedRegistry]);
 
   const fetchRegistries = async () => {
+    let page = 1;
+    const pageSize = 10;
+
     const accessToken = await auth?.srs.getToken();
     const api = new RegistriesApi(
       new Configuration({
         accessToken,
         basePath,
       })
     );
-    await api
-      .getRegistries()
-      .then((res) => {
-        const registry = res?.data;
-        setRegistryItems(registry?.items);
-      })
-      .catch((error) => {
-        //todo: handle error
-      });
+    /**
+     * Api works based on pagination and return 10 records by default.
+     * Getting data by calling api mupltiple times based on page size 10 if total is greater than 10.
+     */
+    const items = await (async (): Promise<Registry[]> => {
+      const response = await api.getRegistries(page, pageSize);
+      let { items } = response?.data;
+      const { total } = response?.data;
+
+      if (total > pageSize) {
```

Comment:
I'm not sure, do we want to show all results without pagination here?

Any particular reason for that?

Wouldn't it be better to fetch more items per page so to do fewer calls? It's better to fetch 100 items in a single fetch than in 10, considering the waterfall and the round trips that adds up quickly

Reply:
This is not table where we have pagination. This data will be populate in dropdown. In real environment user will have limited data but this is our testing org in which we have more than 10 records. 
As backend can't return all the records by default so better to fetch data in more than one api call if we have data more than 10 records.

You should go through jira for more insight about this issue.

Reply:
Thank you, I will. I stand my point tho, one single request asking for more data is better than many requests asking for fewer data.

It's unfortunate the api doesn't allow getting all items at once, but we can ask for 100 items instead of 10 at the very least.

I'd even argue we could use the dropdown with the search and load more functionality, so to keep showing fewer items while giving a way to find what you are looking for. This is if the api allows for filtering.
Anyhow this last bit can be considered a future enhancement. Cc @jgiardino

Reply:
The latter please

Reply:
done. updated page size 100.